### PR TITLE
Alt text images

### LINKED ISSUE #46: Allow for alternative text in non-figure images
We really [should use Alternative text](https://www.imperial.ac.uk/staff/tools-and-reference/web-guide/training-and-events/materials/accessibility/images/) for all our images. But that doesn't play nicely with the MkDocs Caption extension we're using. So I'll approve this for now, but open an issue about sorting this out later.

_Originally posted by @AdrianDAlessandro in https://github.com/ImperialCollegeLondon/RSEBlog/pull/36#discussion_r1721753694_

### PR DESCRIPTION
**Pull request by AdrianDAlessandro**
This PR enables us to use `{: .no-caption }` as an attribute list after defining images for when we do not want there to be a caption.

I will add the following to the README after #41 is merged:

Images are converted to figures automatically using the [MkDocs Caption](https://tobiasah.github.io/mkdocs-caption/) extension. This extension enables putting captions on tables and lists as well as images, however the most basic behaviour uses the image's Alt Text field to populate the figure caption.

For Example:

``` markdown
![figure caption](img.jpg)
```

Will be converted to:

```html
<p>
<figure id=_figure-1>
    <img alt="figure caption" src="img.jpg" />
    <figcaption>Figure 1: figure caption</figcaption>
</figure>
</p>
```

Which will render as a centred image with the caption "Figure 1: figure caption".

If you do not want the image to have a caption, you can label it as a `no-caption` class using an [Attribute List](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#attribute-lists). Reminder, [it is important to include Alt Text](https://www.imperial.ac.uk/staff/tools-and-reference/web-guide/training-and-events/materials/accessibility/images/) for accessibility reasons.

``` markdown
![figure caption](img.jpg) {: .no-caption }
```

You can then add to the attribute list to further specify the style. A useful one is for centering images:

``` markdown
![figure caption](img.jpg) {: .no-caption style="display:block;margin:auto;"}
```

More information on image style options can be found in the [Mozilla Documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#styling_with_css).

Close #46

### CODE REVIEW

**Comment by dc2917** [comment]
This looks good to me, and does what we want it to. 

As a sort-of aside:

> Reminder, it is important to include Alt Text for accessibility.

Would we therefore not want it to do something like:

```html
<p>
<figure id=_figure-1>
    <img src="img.jpg" alt=figure caption/>
    <figcaption>Figure 1: figure caption</figcaption>
</figure>
</p>
```

i.e. populate the `alt` attribute of the `img` tag as well? I'd have thought that if the caption is just text on the page then a screen reader would pick it up, but I had a quick google and a read of [this article](https://www.hassellinclusion.com/blog/figure-figcaption-extended-alternate-text-screen-readers/), where someone has tested different screen readers, browsers and combinations of alt text and caption presence. It seems that in many cases, if if the `alt` attribute is empty, the screen reader just ignores the image, including the caption 😕

**Comment by SaranjeetKaur** [comment]
Thanks @AdrianDAlessandro! I feel it is important to include alt-text on images. Do you think we should link a [alt-text guide](https://github.com/Quansight-Labs/jupyter-accessibility-workshops/blob/fd1d7f96ca40943eda050a339ba64bcf16dd638a/docs/alt-text-guide.md) for reference on the README?

**Comment by AdrianDAlessandro** [comment]
Thanks @dc2917 and @SaranjeetKaur

I've checked the `mkdocs-caption` plugin and it does maintain the alt text (see [source code](https://github.com/tobiasah/mkdocs-caption/blob/ebdc5d7a7f40ad1b52a8954ef8e7a6fcea5f7dd1/src/mkdocs_caption/image.py#L201)), unlike how their documentation suggests (which is where I pulled that lump of html from). So I'll adjust that accordingly for the README.

**Review by dc2917 — APPROVED** [review]

**Review by SaranjeetKaur — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -85,13 +85,46 @@ def add(x, y, z):
...
+```html
+<p>
+<figure id=_figure-1>
+    <img alt="figure caption" src="img.jpg" />
+    <figcaption>Figure 1: figure caption</figcaption>
+</figure>
+</p>
+```
+
+Which will render as a centred image with the caption "Figure 1: figure caption".
```
Comment (dalonsoa) [inline README.md:116]: Maybe mention that it also becomes the actual Alt Text, in addition to the caption, important for screen readers, as mentioned?

**Review by dalonsoa — APPROVED** [review]
LGTM!

**Pull request merged**